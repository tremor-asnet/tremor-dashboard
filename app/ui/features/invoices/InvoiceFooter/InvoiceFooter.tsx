"use client";

//Components
import { Button, Text, Flex } from "@tremor/react";

//Styles
import "@/styles/billing.css";

const InvoiceFooter = () => {
  const handlePrintInvoice = () => {
    window.print();
  };

  return (
    <Flex
      alignItems="start"
      flexDirection="col"
      className="md:items-end md:flex-row mt-10 md:mt-20 invoice">
      <Flex
        flexDirection="col"
        alignItems="start"
        className="print-footer md:max-w-[40%]">
        <Text className="text-primary dark:text-white text-xl font-semibold leading-7 dark:print:text-white">
          Thank you!
        </Text>
        <Text className="dark:text-dark-romance font-light text-secondary leading-5 my-2 dark:print:text-lighter">
          If you encounter any issues related to the invoice you can contact us
          at:
        </Text>
        <Text className="text-tremor-title dark:text-dark-romance text-secondary leading-5 mt-4 dark:print:text-lighter">
          email:{" "}
          <a
            className="text-primary dark:text-white dark:print:text-white"
            href="mailto:support@creative-tim.com">
            support@creative-tim.com
          </a>
        </Text>
      </Flex>
      <Button
        className="py-3 px-6 mt-6 md:mt-0 bg-gradient-primary dark:bg-gradient-pickled border-none dark:text-white print:hidden"
        onClick={handlePrintInvoice}>
        <Text className="uppercase font-bold text-xs text-white dark:text-white tracking-wide">
          print
        </Text>
      </Button>
    </Flex>
  );
};

export default InvoiceFooter;
