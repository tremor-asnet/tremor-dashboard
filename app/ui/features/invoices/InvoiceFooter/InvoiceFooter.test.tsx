import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";

// Components
import InvoiceFooter from "./InvoiceFooter";

describe("InvoiceFooter Testing", () => {
  it("should match snapshot", () => {
    const { container } = render(<InvoiceFooter />);

    expect(container).toMatchSnapshot();
  });

  it("prints invoice when print button is clicked", () => {
    window.print = jest.fn();

    const { getByRole } = render(<InvoiceFooter />);

    fireEvent.click(
      getByRole("button", {
        name: /print/i,
      }),
    );

    expect(window.print).toHaveBeenCalled();
  });
});
