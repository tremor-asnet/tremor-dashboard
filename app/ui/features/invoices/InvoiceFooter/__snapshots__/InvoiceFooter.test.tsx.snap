// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceFooter Testing should match snapshot 1`] = `
<div>
  <div
    class="tremor-Flex-root flex w-full flex-col justify-between items-start md:items-end md:flex-row mt-10 md:mt-20 invoice"
  >
    <div
      class="tremor-Flex-root flex w-full flex-col justify-between items-start print-footer md:max-w-[40%]"
    >
      <p
        class="text-tremor-default text-primary dark:text-white text-xl font-semibold leading-7 dark:print:text-white"
      >
        Thank you!
      </p>
      <p
        class="text-tremor-default dark:text-dark-romance font-light text-secondary leading-5 my-2 dark:print:text-lighter"
      >
        If you encounter any issues related to the invoice you can contact us at:
      </p>
      <p
        class="text-tremor-title dark:text-dark-romance text-secondary leading-5 mt-4 dark:print:text-lighter"
      >
        email:
         
        <a
          class="text-primary dark:text-white dark:print:text-white"
          href="mailto:support@creative-tim.com"
        >
          support@creative-tim.com
        </a>
      </p>
    </div>
    <button
      class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input border text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis py-3 px-6 mt-6 md:mt-0 bg-gradient-primary dark:bg-gradient-pickled border-none dark:text-white print:hidden"
    >
      <span
        class="tremor-Button-text text-sm whitespace-nowrap"
      >
        <p
          class="text-tremor-default uppercase font-bold text-xs text-white dark:text-white tracking-wide"
        >
          print
        </p>
      </span>
    </button>
  </div>
</div>
`;
